salvo/src/lib/benchmark/fully_dockerized_benchmark.py

```diff
@@ -130,11 +131,11 @@ def execute_benchmark(self) -> None:
     }
     log.debug(f"Using environment: {image_vars}")
 
-    volumes = get_docker_volumes(output_dir, test_dir)
+    volumes = base_benchmark.get_docker_volumes(output_dir, test_dir)
     log.debug(f"Using Volumes: {volumes}")
-    self.set_environment_vars()
+    self._set_environment_vars()
```

Comment:
(optional, for a future PR) This pattern where we need to do something and then clear something up tends to be brittle in Python when written like this. Say we have a pattern like this:

```python
self._set_environment_vars()

some_code

self._clear_environment_vars()

```

We can find two ways how this can get broken and end up not clearing the variables:

1. If **some_code** raises any sort of exception, the execution terminates immediately and `self._clear_environment_vars()` never gets called.
1. If a future code change adds a new return statement, a bug is born. The developer developer adding the new return might easily miss the need to cleanup. An example of such a change:

```python
self._set_environment_vars()

some_code
if some_condition:
  return # Exits without clearing.

self._clear_environment_vars()

```

Python's [data model](https://docs.python.org/3/reference/datamodel.html) has an answer for this problem - the concept of a [context manager](https://docs.python.org/3/reference/datamodel.html#with-statement-context-managers) that can be used in a [with statement](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement), changing our code to this form:

```python
with some_environment_controller:
  some_code
```

To achieve this, we would define a new object specifically responsible for setting and clearing the environment variables and implement methods `__enter__` and `__exit__` on them as described [here](https://docs.python.org/3/reference/datamodel.html#with-statement-context-managers). Essentially the `self._set_environment_vars()` method would become `__enter__` and the `self._clear_environment_vars()` method would become `__exit__`. 

This solves both the problems, because the `__exit__` method is called even if exceptions are raised and it is also impossible to add early returns that would bypass `__exit__` as long as the code continues to utilize the with statement.

Reply:
Ack.  Will work on this.